daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetImpl.scala

```diff
@@ -840,20 +840,9 @@ sealed trait DIElement
     case _ => false
   }
 
-  def toRootDoc: DIDocument = toRootDoc1(this)
-
-  private def toRootDoc1(orig: DIElement): DIDocument = {
+  final lazy val toRootDoc: DIDocument = {
     if (isRootDoc) this.asInstanceOf[DIDocument]
-    else if (isRoot) diParent.asInstanceOf[DIDocument]
-    else {
-      parent match {
-        case null =>
-          orig.erd.toss(new InfosetNoRootException(orig, erd))
-        case elt: DIElement => elt.toRootDoc1(orig)
-        case _ =>
-          orig.erd.toss(new InfosetNoRootException(orig, erd))
-      }
-    }
+    else toParent.toRootDoc
```

Comment:
Did you intended to make this a lazy val? I think the other review mentioned keeping a def unless it came up in profiling?

Reply:
Force of habit. I can make this a def. But first I'm going to check if it is even used anymore and remove it if it is not.

Reply:
This is used exactly one place in the runtime. When an expression uses an absolute path, then this is invoked to get back up to the root. 

This can be done differently, just walking up to the deepest element on the currentNodeStack in pstate/ustate, so I'm going to suggest we should drop this method entirely.